src/status_im/contexts/wallet/add_account/add_address/view.cljs

```diff
@@ -7,28 +7,38 @@
     [reagent.core :as reagent]
     [status-im.common.floating-button-page.view :as floating-button-page]
     [status-im.constants :as constants]
-    [status-im.contexts.wallet.add-account.add-address-to-watch.style :as style]
+    [status-im.contexts.wallet.add-account.add-address.style :as style]
     [status-im.contexts.wallet.common.validation :as validation]
     [status-im.subs.wallet.add-account.address-to-watch]
     [utils.debounce :as debounce]
     [utils.i18n :as i18n]
     [utils.re-frame :as rf]))
 
+(def ^:private adding-addresses-purposes
+  {:watch {:title       :t/add-address-to-watch
+           :description :t/enter-eth
+           :input-title :t/eth-or-ens}
+   :save  {:title       :t/add-address
+           :description :t/save-address-description
+           :input-title :t/address-or-end-name}})
+
 (defn- validate-address
-  [known-addresses user-input]
+  [known-addresses user-input purpose]
   (cond
-    (or (nil? user-input) (= user-input "")) nil
-    (contains? known-addresses user-input)   (i18n/label :t/address-already-in-use)
+    (or (nil? user-input) (= user-input ""))     nil
```

Comment:
`(or (empty? user-input)) nil`?

Reply:
`empty?` doesn't cater for whitespace, I think it shouldn't be used for strings often

Reply:
you can use `blank?` of course, but that would be changing the functionality, not sure what the original intention of the code was, `empty?` should retain it, while blank adds some extra cases, but either are fine by me, but `or` looks unneccessary